forensics/disk_image.py

```
@@ -0,0 +1,104 @@
+
+    # Try a partition table first; fall back to treating the image as a single
+    # filesystem (common for extracted single-volume images).
+    filesystems = []
+    try:
+        volume = pytsk3.Volume_Info(img)
+        for part in volume:
+            if part.len > 0 and b"Unallocated" not in part.desc:
+                try:
+                    filesystems.append(pytsk3.FS_Info(img, offset=part.start * volume.info.block_size))
+                except Exception:
+                    continue
```

Ruff (0.16.2)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)